### PR TITLE
[WIP] Converge generateCmd and deployCmd implementations

### PR DESCRIPTION
**Pull request by serbrech**
**What this PR does / why we need it**:
`generateCmd` and `deployCmd` have a lot of duplicated code. `Deploy` works by first generating the ARM templates, and then automating the azure deployment. This PR is an attempt at converging the implementations to reduce code duplication and ease maintenance.

**Special notes for your reviewer**:
This is WIP, but please review the direction already.  
Converging deploy and generate has for consequence that, it adds the _default apimodel_ functionality to the generate command. Avoiding this was more work that sharing it for both commands.

**If applicable**:
- [x] converge commands parameters
- [x] adjust unit tests
- [x] reuse mergeAPIModel implementation
- [x] reuse generateCmd validation
- [ ] ~~reuse the loadApiModel implementation :~~ most likely in a separate PR as it needs some refactoring in apiloader for validation
- [ ] improve test coverage
- [ ] documentation where necessary 

**Release note**:

```release-note
```

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -112,24 +112,9 @@ func newDeployCmd() *cobra.Command {
 func (dc *deployCmd) validateArgs(cmd *cobra.Command, args []string) error {
 	var err error
 
-	dc.locale, err = i18n.LoadTranslations()
```
Comment (jackfrancis) [inline cmd/deploy.go]: Why are we getting rid of this? The `loadAPIModel` method is still using this private property:

```
apiloader := &api.Apiloader{
 		Translator: &i18n.Translator{
 			Locale: dc.locale,
 		},
 	}
```

Comment (serbrech) [reply cmd/deploy.go]: it's done in the `dc.validate()`, which is really `dc.generateCmd.validate()` just below.
it should probably be moved out of validate and into the creation of the cmds.

**Comment by serbrech** [comment]
rebased

**Comment by serbrech** [comment]
Obsolete, will attempt at a later stage

**Pull request closed without merging**